src/components/App.jsx

```diff
@@ -29,10 +37,42 @@ class App extends React.Component {
     if (!props.initialised) {  //NOTE: This should almost always trigger, since App.constructor() triggers exactly once, on the website loading, when all initial values are at their default.
       props.dispatch(checkLoginUser());
     }
+    
+    //SPECIAL: users on ASM tend to stay in a single session for WAY longer
+    //than a standard Zooniverse CFE user, so we're encountering issues where
+    //their login tokens timeout in the middle of classifying a document.
+    //The following is a mechanism that checks if a user is logged in before
+    //performing any action that may require a valid login token, taking safety
+    //actions if necessary.
+    apiClient.beforeEveryRequest = this.checkIfLoggedInUserIsStillLoggedIn.bind(this);
   }
-
-  returnSometrolhing(something) { // eslint-disable-line class-methods-use-this
-    return something;
+  
+  checkIfLoggedInUserIsStillLoggedIn() {
+    const props = this.props;
+    
+    return oauth.checkBearerToken()
+      .then((token) => {
+        console.log(`checkBearerToken - initialised: ${!!props.initialised}, user: ${!!props.user}, token: ${!!token}`);
```

Comment:
I think it would be good at some point, perhaps after this login issue is resolved, to remove the logs we have in the app that were initially used to pinpoint this bug.

Reply:
Thanks for spotting this - this console.log (and any others I missed) should have been marked with that //DEBUG/TEST tag so I could Ctrl+F and remove 'em in the follow up cleanup.